packages/sdk-router/src/sdk.ts

```diff
@@ -39,141 +53,396 @@ class SynapseSDK {
       `Amount of chains and providers does not equal`
     )
     this.synapseRouters = {}
+    this.synapseCCTPRouters = {}
     this.providers = {}
     for (let i = 0; i < chainIds.length; i++) {
       this.synapseRouters[chainIds[i]] = new SynapseRouter(
         chainIds[i],
         providers[i]
       )
       this.providers[chainIds[i]] = providers[i]
+      // check if the chain id is in the CCTP_ROUTER_ADDRESS object
+      if(CCTP_ROUTER_ADDRESS.hasOwnProperty(chainIds[i])) {
+        this.synapseCCTPRouters[chainIds[i]] = new SynapseCCTPRouter(
+          chainIds[i],
+          providers[i]
+        )
+      }
     }
   }
 
-  public async bridgeQuote(
-    originChainId: number,
+
+  public async getBridgeTokens(
     destChainId: number,
-    tokenIn: string,
     tokenOut: string,
-    amountIn: BigintIsh,
-    deadline?: BigNumber
-  ): Promise<{
-    feeAmount: BigNumber | undefined
-    feeConfig: FeeConfig | undefined
-    routerAddress: string | undefined
-    maxAmountOut: BigNumber | undefined
-    originQuery: Query | undefined
-    destQuery: Query | undefined
-  }> {
-    tokenOut = handleNativeToken(tokenOut)
-    tokenIn = handleNativeToken(tokenIn)
-    const originRouter: SynapseRouter = this.synapseRouters[originChainId]
-    const destRouter: SynapseRouter = this.synapseRouters[destChainId]
-    const routerAddress = originRouter.routerContract.address
-
-    let bridgeTokens = this.bridgeTokenCache[destChainId + '_' + tokenOut]
-    if (!bridgeTokens) {
-      const routerBridgeTokens =
-        await destRouter.routerContract.getConnectedBridgeTokens(tokenOut)
-
-      // Filter tokens with a valid symbol and address
-      bridgeTokens = routerBridgeTokens.filter(
-        (bridgeToken) =>
-          bridgeToken.symbol.length && bridgeToken.token !== AddressZero
-      )
-
-      // Throw error if no valid bridge tokens found
-      if (!bridgeTokens?.length) {
-        throw new Error('No bridge tokens found for this route')
+    destRouters: (SynapseRouter | SynapseCCTPRouter)[]
+  ): Promise<{ symbol: string; token: string }[][]> {
+
+    const bridgeTokensPromises = destRouters.map(async (destRouter) => {
+      // Check the cache first
+      let bridgeTokens = this.bridgeTokenCache[destChainId + '_' + tokenOut];
```

Comment:
The cache key should include the router name (or simply address) as well, otherwise this creates race conditions

Reply:
Resolved in https://github.com/synapsecns/sanguine/pull/1059/commits/59f46fb19a1f6249b3ba41632239f55f0fbf8135